docs/spec/identities.md

```diff
@@ -0,0 +1,438 @@
+      revision is known
+    * **Or**, the set of signatures forms a quorum of the delegations of the
+      **previous** revision.
+
+    It is an error if:
+
+    * No previous revision is given, but a parent in the identity attestation
+      chain is found
+    * A previous revision is given, but the identity attestation chain does not
+      yield a parent
+    * A previous revision is given, but it is not the same the parent
+      attestation refers to
+    * The current and parent attestations refer to different `root`s
+
+The verification process can now be described recursively:
+
+```rust
+/* State transitions, definitions elided for brevity */
+fn signed(i: Untrusted<Identity>) -> Result<Signed<Identity>, Error>;
+fn quorum(i: Signed<Identity>) -> Result<Quorum<Identity>, Error>;
+fn verified(i: Quorum<Identity>, parent: Option<Verified<Identity>>)
+    -> Result<Verified<Identity>, Error>;
+
+fn verify(
+    head: Untrusted<Identity>,
+    mut parents: impl Iterator<Item = Untrusted<Identity>>,
+) -> Result<Verified<Identity>, Error> {
+    let head = quorum(signed(head)?)?;
+    let parent = match parents.next() {
+        None => Ok(None),
+        Some(parent) => verify(parent, parents),
+    }?;
+
+    verified(head, parent)
+}
+```
+
+> **Implementation Notes**:
+>
+> * The recusive definition is given for brevity, implementations may prefer to
+>   walk the history in reverse order.
+> * As an optimisation, implementations SHOULD store the result of a successful
+>   verification in persistent storage, such that verification of updates can
+>   start from the last-good state instead of the history root. Observe,
+>   however, that two different `Identity` objects (as identitified by their
+>   `id`) may attest the same `Doc` revision -- the persistent state SHOULD
+>   therefore allow retrieving the set of attesting `Identity`s by `Doc`
+>   revision.
+> * The `quorum` predicate above SHOULD actually skip over this parent instead
+>   of aborting the procedure -- it is possible that a later `Identity` reaches
+>   the quorum.
+
+## Delegations
+
+Radicle Link distinguishes two types of identities: personal and project. The
+first describes an actor in the system, while the second describes a (software)
+project on which one or more actors collaborate. Apart from their payload types
+`T`, they differ in their delegations type `D`:
+
+Personal identities can only delegate to anonymous keys, while project
+identities MAY attach a personal identity (by it's URN) to a key delegation.
+
+More formally:
+
+```rust
+type User<T> = Identity<T, HashSet<PublicKey>>;
+type Project<T> = Identity<T, HashMap<PublicKey, Option<Urn>>>;
+```
+
+> **TODO** Consider a container for project delegations which does not require
+> to repeat the URN for every key
+
+The set of `PublicKey`s delegated to MUST be unique per identity document.
+
+The purpose of the project-user link is to allow applications to display
```

Comment:
I thought the purpose of the maintainer list was to establish who is trusted to operate on the project itself (not just on the project identity document but on the actual project).
Like, who can sign tags, or new releases... in other words, who is a trusted maintainer.
The ability to display additional metadata should be a side effect of knowing the set of maintainers.